lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp

```diff
@@ -220,8 +231,16 @@ struct DotOpMFMAConversionHelper {
         return MatrixCoreType::FP32_BF16_BF16_FP32;
       }
     }
-    if (aElemTy.isInteger(8))
-      return MatrixCoreType::INT32_INT8_INT8_INT32;
+    if (aElemTy.isInteger(8)) {
+      auto nonKDim = mfmaEncoding.getNonKDim();
+      auto kWidth = dotOpEncoding.getKWidth();
+      if ((nonKDim == 32 && kWidth == 8) || (nonKDim == 16 && kWidth == 8)) {
```

Comment:
You can only check kWidth here

Reply:
I think keeping `nonKDim` checks will not hurt.
This is like an additional sanity check.